lib/parsec/util.py

```diff
@@ -26,6 +26,75 @@
 from parsec.OrderedDict import OrderedDictWithDefaults
 
 
+def intlistjoin(lst):
+    """Return dump string for int list.
+
+    Attempt grouping on sequences with 3 or more numbers:
+    * Consider sequential int values in list, group them together in
+      `START..END[..STEP]` syntax where relevant.
+    * Consider same numbers in list, group them together in `N*INT` syntax.
+
+    Arguments:
+        lst (list): a list of int numbers.
+
+    Return (str):
+        The (hopefully) nicely formatted dump string.
+
+    Examples:
+        >>> intlistjoin([])
+        ''
+        >>> intlistjoin([10])
+        '10'
+        >>> intlistjoin([10, 10])
+        '10, 10'
+        >>> intlistjoin([10, 10, 10])
+        '3*10'
+        >>> intlistjoin([10, 11])
+        '10, 11'
+        >>> intlistjoin([10, 11, 12])
+        '10..12'
+        >>> intlistjoin([-1, 1, 3, 5, 0, 0, 0, 8, 9, 11, -1, 0, 1, 2])
+        '-1..5..2, 3*0, 8, 9, 11, -1..2'
+        >>> intlistjoin([-1, 1, 3, 5, 0, 0, 0, 8, 9, 11, 11, 12])
+        '-1..5..2, 3*0, 8, 9, 11, 11, 12'
+        >>> intlistjoin(
+        ...     [-10, -10, -1, 1, 3, 5, 0, 0, 0, 8, 9, 11, -1, 0, 1, 2])
+        '-10, -10, -1..5..2, 3*0, 8, 9, 11, -1..2'
+        >>> intlistjoin(
+        ...     [64747, -10, -10, -10, -1, 1, 3, 5, 0, 0, 0, 8, 9, 11, -1, 0,
+        ...      1, 2, 3, 4, 19, 19, 19, 20, 20, 20, 21, 22, 23])
+        '64747, 3*-10, -1..5..2, 3*0, 8, 9, 11, -1..4, 3*19, 3*20, 21..23'
+    """
+    rets = []
+    items = list(lst)
```

Comment:
```suggestion
    items = list(sorted(lst))
```

```python
>>> intlistjoin([4,5,6,1,2,3])  # before
'4..6, 1..3'
>>> intlistjoin(sorted([4,5,6,1,2,3]))  # after
'1..6'

>>> intlistjoin([0,1,0,1,0,1])  # before
'0, 1, 0, 1, 0, 1'
>>> intlistjoin(sorted([0,1,0,1,0,1]))  # after
'3*0, 3*1'
```

Reply:
Unfortunately, order should be preserved for a list.